modules/module-cloudwatch/variables.tf

```diff
@@ -22,3 +22,17 @@ variable "use_cloudwatch_for_logging" {
   description = "boolean variable to enable the cloudwatch usage for logs storage"
   default     = false
 }
+
+variable "use_cloudwatch_for_monitoring" {
+  description = "boolean variable to enable the cloudwatch usage for instance monitoring"
+  default     = false
+}
+
+variable "ec2_instance_id" {
```

Comment:
This is only known after the first apply right?

Reply:
Why is this only here but not on the main variables file?

Reply:
In the first application, terraform will check the dependency between modules and will execute the module-ec2-linux-web  (EC2) module before the cloudwatch one.
The `ec2_instance_id` isn't required in the main variables because the value is get from the `module-ec2-linux-web` output.
Line 108 in the main file:
```
ec2_instance_id                 = module.module-ec2-linux-web.server_id
```